runtime-modules/project-token/src/lib.rs

```diff
@@ -0,0 +1,686 @@
+        });
+
+        let new_yearly_rate = new_rate.to_yearly_rate_representation(T::BlocksPerYear::get());
+        Self::deposit_event(RawEvent::PatronageRateDecreasedTo(
+            token_id,
+            new_yearly_rate,
+        ));
+
+        Ok(())
+    }
+
+    /// Allow creator to receive credit into his accounts
+    /// Preconditions:
+    /// - `token_id` must exists
+    /// - `to_account` must be valid for `token_id`
+    ///
+    /// Postconditions:
+    /// - outstanding patronage credit for `token_id` transferred to `to_account`
+    /// - outstanding patronage credit subsequently set to 0
+    /// no-op if outstanding credit is zero
+    fn claim_patronage_credit(token_id: T::TokenId, to_account: T::AccountId) -> DispatchResult {
+        let token_info = Self::ensure_token_exists(token_id)?;
+        Self::ensure_account_data_exists(token_id, &to_account).map(|_| ())?;
+
+        let now = Self::current_block();
+        let unclaimed_patronage = token_info.unclaimed_patronage_at_block(now);
+
+        if unclaimed_patronage.is_zero() {
+            return Ok(());
+        }
+
+        // == MUTATION SAFE ==
+
+        AccountInfoByTokenAndAccount::<T>::mutate(token_id, &to_account, |account_info| {
+            account_info.increase_liquidity_by(unclaimed_patronage)
+        });
+
+        TokenInfoById::<T>::mutate(token_id, |token_info| {
+            token_info.increase_supply_by(unclaimed_patronage);
+            token_info.set_unclaimed_tally_patronage_at_block(T::Balance::zero(), now);
+        });
+
+        Self::deposit_event(RawEvent::PatronageCreditClaimed(
+            token_id,
+            unclaimed_patronage,
+            to_account,
+        ));
+
+        Ok(())
+    }
+
+    /// Issue token with specified characteristics
+    /// Preconditions:
+    /// - `token_id` must NOT exists
+    /// - `symbol` specified in the parameters must NOT exists
+    ///
+    /// Postconditions:
+    /// - token with specified characteristics is added to storage state
+    /// - `NextTokenId` increased by 1
+    /// - symbol is added to `Symbols`
+    fn issue_token(
+        owner_account_id: T::AccountId,
+        issuance_parameters: TokenIssuanceParametersOf<T>,
+    ) -> DispatchResult {
+        let token_id = Self::next_token_id();
+        Self::validate_issuance_parameters(&issuance_parameters)?;
+
+        let now = Self::current_block();
+        let initial_supply = issuance_parameters.initial_supply;
+        let token_data = issuance_parameters.build::<_, T>(now);
+
+        // == MUTATION SAFE ==
+
+        SymbolsUsed::<T>::insert(&token_data.symbol, ());
+        TokenInfoById::<T>::insert(token_id, token_data.clone());
+        // AFTER token is added to storage add account
+        Self::do_insert_new_account_for_token(
+            token_id,
+            &owner_account_id,
+            AccountDataOf::<T>::new_with_liquidity_and_bond(initial_supply, BloatBond::<T>::get()),
```

Comment:
The bloat bond is not actually transferred to module account, so please:
- Add a check before `== MUTATION SAFE ==` (make sure issuer has enough funds to cover the bloat bond)
- Add a transfer from issuer's account into module account under `== MUTATION SAFE ==`
- Add tests to verify that the bloat bond during `issue_token` works as expected

Reply:
fixed @ 51ee247b569fb4ccf0bd2bd43a22b1e4dc11333a